factoryWithTypeCheckers.js

```diff
@@ -10,6 +10,7 @@
 var emptyFunction = require('fbjs/lib/emptyFunction');
 var invariant = require('fbjs/lib/invariant');
 var warning = require('fbjs/lib/warning');
+var has = require('has');
```

Comment:
Can we avoid taking on more npm deps please? Just inline it. Thanks!

Reply:
In that case, `Function.call.bind(Object.prototype.hasOwnProperty)`